### ISSUE TITLE
no-member reported when accessing a @property of parent class

### ISSUE THREAD

**Issue by jscn** [PyCQA/pylint#3531]
I'm not sure if this bug is in `pylint`, `astroid` or `WTForms`. I am having trouble reproducing it with similar code to what is in `WTForms`.

`self.data` in this case is a reference [to a method](https://github.com/wtforms/wtforms/blob/master/src/wtforms/form.py#L137) which has been decorated with `@property` on the `wtforms.form.BaseForm` class (from which `wtforms.form.Form` inherits).

### Steps to reproduce
1. Install `WTForms` and create a sub-class of the `Form` class which accesses `self.data` (call it `BadForm` for argument's sake).
2. Run pylint against the class

### Current behavior
The following error is reported: `E1101: Instance of 'BadForm' has no 'data' member (no-member)`

### Expected behavior
No `no-member` error is reported.

### pylint --version output
```
astroid==2.4.0
isort==4.3.21
lazy-object-proxy==1.4.3
MarkupSafe==1.1.1
mccabe==0.6.1
pylint==2.5.0
six==1.14.0
toml==0.10.0
typed-ast==1.4.1
wrapt==1.12.1
WTForms==2.3.1
```
Below is some code demonstrating the problem, and my attempts to reproduce it without using `WTForms` directly:
```python
"""Demonstrate false positive for no-member."""
from wtforms.form import BaseForm, FormMeta, Form
from wtforms.meta import DefaultMeta


# pylint: disable=too-few-public-methods,missing-function-docstring,missing-class-docstring,too-many-arguments


# This demonstrates the exact error I am seeing:
class BadForm(Form):
    """A class demonstrating the issue that appears when inheriting from `Form`."""

    def do_something_with_data(self):
        return self.data


# This following two classes demonstrate the @property access working without an error.
class ClassWithProperty:
    """A class with a property."""

    @property
    def a_property(self):
        return "this works"


class AccessPropertyOk(ClassWithProperty):
    """A class that accesses a property on its parent."""

    def do_something(self):
        return self.a_property


# It doesn't seem to have anything to do with how many classes deep the inheritance chain is:
class IsItBecauseOfSeveralClassesInTheHierarchy(AccessPropertyOk):

    def something_else(self):
        return self.a_property


# The error doesn't appear when inheriting from the direct parent of `wtforms.forms.Form`:
class BaseFormChild(BaseForm):

    def do_something_with_data(self):
        return self.data


# Ok, so lets try re-implementing the hierarchy from `WTForms`. This
# is basically what is in `wtforms.forms.Form`
class FormWithMetaclassAndMeta(BaseForm, metaclass=FormMeta):

    Meta = DefaultMeta

    def do_something_with_data(self):
        return self.data


# I don't understand why this does not report an error. I can't see
# how this differs from `BadForm` in any material way:
class ChildForm(FormWithMetaclassAndMeta):

    def something(self):
        return self.data
```

**Comment by Hubrrr88** [comment]
I have a false-positive for this issue with a very reduced code, which is still a bug in this pylint version:
```
pylint 2.15.10
astroid 2.13.2
Python 3.9.5 (default, Oct  6 2021, 16:12:40)
[Clang 13.0.0 (clang-1300.0.29.3)]
```

Code to reproduce the mentioned issue above:
```
class Interface:
    def interface_function(self):
        print(type(self))


class Parent:
    __slots__ = ("interface",)

    def __init__(self, interface: Interface) -> None:
        self.interface: Interface = interface

    @property
    def property1(self) -> Interface:
        return self.interface

    @property1.setter
    def property1(self, value: Interface) -> None:
        self.interface = value


class Child(Parent):
    __slots__ = ()

    @Parent.property1.setter
    def property1(self, value: Interface) -> None:
        # Calling super setter
        Parent.property1.fset(self, value)
        # super(Child, Child).property1.__set__(self, value)


class ChildChild(Child):
    __slots__ = ()

    def __init__(self, interface: Interface) -> None:
        super().__init__(interface)
        self.property1.interface_function()

    def new_function(self):
        self.property1.interface_function()


class InterfaceImpl2(Interface):
    pass


if __name__ == '__main__':
    cc = ChildChild(Interface())
    cc.new_function()
    cc.property1 = InterfaceImpl2()
    cc.new_function()
```

Code runs fine, and the pylint error is gone if the setter is not overloaded and using fset to call the super setter of the base class.

Should I open a new issue for this simple reproducible false-positive?

**Comment by DanielNoord** [comment]
I'd say for now this issue is enough. If it turns out to be a unrelated issue we can open a new one! Thanks for considering it though and taking the time to look for duplicate issues!
